language_tour.md

```diff
@@ -366,6 +366,39 @@ The patterns are not exhaustive. Missing cases:
        | }
 ```
 
+### For-Each-Join Loops
+
+Garble has special support for joining together two sorted arrays of tuples, by comparing their first field for equality, which can be useful to combine two data sources coming from different parties similar to a JOIN in SQL. Syntactically for-each-join loops are a special case of for-each loops, using a `join` function instead of an array:
+
+```rust
+let rows1 = [(0u8, 10u16), (1u8, 11u16), (2u8, 12u16)];
+let rows2 = [(0u8, 5u32, 5u32), (2u8, 6u32, 6u32)];
+// The arrays rows1 and rows2 will be joined based on their first field, which is of type u8.
+// The tuple (1u8, 11u16) from rows1 is discarded because it cannot be joined with rows2.
+for joined in join(rows1, rows2) {
+    let ((id1, x), (id2, y, z)) = joined;
+    // ...
+}
+```
+
+Garble automatically joins the arrays in a for-each-join loop using a [bitonic sorting network](https://en.wikipedia.org/wiki/Bitonic_sorter), more concretely implementing the paper [Private Set Intersection:
+Are Garbled Circuits Better than Custom Protocols?](https://www.ndss-symposium.org/wp-content/uploads/2017/09/06_4.pdf), which has a circuit complexity of `(m + n) * log(m + n)` instead of `m * n` which would result from joining the arrays using nested loops. **The arrays that are joined in the loop must be sorted**, otherwise elements might be discarded or invalid data returned.
```

Comment:
What we could also add is that we do not do the shuffle step of their protocol as the join is intended to be used further in computation. But if someone only would want to do join and reveal the result that would still need to be implemented?

Reply:
Good point, I updated it slightly.